src/Authentication/Authentication/Utilities/DependencyAssemblyResolver.cs

```diff
@@ -0,0 +1,101 @@
+﻿// ------------------------------------------------------------------------------
+//  Copyright (c) Microsoft Corporation.  All Rights Reserved.  Licensed under the MIT License.  See License in the project root for license information.
+// ------------------------------------------------------------------------------
+
+namespace Microsoft.Graph.PowerShell.Authentication.Utilities
+{
+    using System;
+    using System.Collections.Generic;
+    using System.IO;
+    using System.Reflection;
+    public static class DependencyAssemblyResolver
+    {
+        // Catalog our dependencies here to ensure we don't load anything else.
+        private static IReadOnlyDictionary<string, Version> Dependencies = new Dictionary<string, Version>
+        {
+            { "Microsoft.Identity.Client", new Version("4.23.0.0") },
```

Comment:
Can the resolution logic use a file instead of being explicitly set here?

For example read the csproj to determine this list?

Reply:
We can't access csproj here since this is done at runtime and not build time. We need to know which assembly and version to load on our own when PowerShell imports a module thus the need for an in-memory dictionary.